Code/Source/GNSS/ROS2GNSSSensorComponent.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "GNSS/ROS2GNSSSensorComponent.h"
+#include "Frame/ROS2FrameComponent.h"
+#include "ROS2/ROS2Bus.h"
+#include "Utilities/ROS2Conversions.h"
+#include "Utilities/ROS2Names.h"
+#include "AzCore/Math/Matrix4x4.h"
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/EditContextConstants.inl>
+
+namespace ROS2
+{
+    namespace Internal
+    {
+        const char* kGNSSMsgType = "geometry_msgs::msg::PoseStamped";
```

Comment:
Should be https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/NavSatFix.msg
as well as https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/NavSatStatus.msg

The pose can be used as well, but needs explanation on what the data is.